deploying enterprise server on amazon without --credentials flag crashes pachctl

**What happened?**:

```
$ pachctl  deploy amazon ${AWS_REGION} ${STORAGE_SIZE}  --dynamic-etcd-nodes 1  --enterprise-server --dry-run
WARNING: The version of Pachyderm you are deploying (2.0.0-beta.1) is an unstable pre-release build and may not support data migration.

Are you sure you want to do this? (y/N): y
got confirm result: y

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x23bb963]

goroutine 1 [running]:
github.com/pachyderm/pachyderm/v2/src/internal/deploy/assets.WriteAmazonAssets(0x308a8c8, 0xc000d73fb0, 0xc00011da00, 0x7ffeefbff5c8, 0x9, 0x0, 0x0, 0xa, 0x0, 0x0, ...)
	src/internal/deploy/assets/assets.go:1058 +0xa3
github.com/pachyderm/pachyderm/v2/src/internal/deploy/cmds.standardDeployCmds.func9(0xc0000b2900, 0x2, 0x6, 0x0, 0x0)
	src/internal/deploy/cmds/cmds.go:716 +0x674
github.com/pachyderm/pachyderm/v2/src/internal/cmdutil.RunBoundedArgs.func1(0xc0003e1680, 0xc0000b2900, 0x2, 0x6)
	src/internal/cmdutil/cobra.go:65 +0x185
github.com/spf13/cobra.(*Command).execute(0xc0003e1680, 0xc0000b2840, 0x6, 0x6, 0xc0003e1680, 0xc0000b2840)
	/Users/nitin/go/pkg/mod/github.com/spf13/cobra@v0.0.6-0.20191202130430-b04b5bfc50cb/command.go:833 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0xc0001ed900, 0x3ea464c, 0xc000c5fef8, 0x14cc185)
	/Users/nitin/go/pkg/mod/github.com/spf13/cobra@v0.0.6-0.20191202130430-b04b5bfc50cb/command.go:917 +0x30b
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/nitin/go/pkg/mod/github.com/spf13/cobra@v0.0.6-0.20191202130430-b04b5bfc50cb/command.go:867
main.main.func1(0x0, 0x0)
	src/server/cmd/pachctl/main.go:20 +0x51
main.main()
	src/server/cmd/pachctl/main.go:21 +0xbc

```

**What you expected to happen?**:

It should have told me that --credentials was required.

**How to reproduce it (as minimally and precisely as possible)?**:

See above

**Anything else we need to know?**:

**Environment?**:
- Kubernetes version (use `kubectl version`):
- Pachyderm CLI and pachd server version (use `pachctl version`):
- Cloud provider (e.g. aws, azure, gke) or local deployment (e.g. minikube vs dockerized k8s):
- OS (e.g. from /etc/os-release):
- Others:

--- a/src/internal/deploy/assets/assets.go
+++ b/src/internal/deploy/assets/assets.go
@@ -1056,7 +1056,7 @@ func WriteAmazonAssets(encoder serde.Encoder, opts *AssetOpts, region string, bu
 		return err
 	}
 	var secret map[string][]byte
-	if creds.ID != "" {
+	if creds != nil && creds.ID != "" {
 		secret = AmazonSecret(region, bucket, creds.ID, creds.Secret, creds.Token, cloudfrontDistro, "", advancedConfig)
 	}
 	return WriteSecret(encoder, secret, opts)

--- a/src/internal/deploy/cmds/cmds.go
+++ b/src/internal/deploy/cmds/cmds.go
@@ -639,7 +639,7 @@ If <object store backend> is \"s3\", then the arguments are:
 			// Require credentials to access S3 for pachd deployments.
 			// Enterprise server deployments don't require an S3 bucket, so they don't need credentials.
 			if creds == "" && !enterpriseServer {
-				return errors.Errorf("one of --credentials, or --iam-role needs to be provided")
+				return errors.Errorf("--credentials needs to be provided")
 			}
 
 			// populate 'amazonCreds' & validate

